.github/workflows/commit.yml

```diff
@@ -0,0 +1,58 @@
+name: commit
+on: push
+jobs:
+  build_and_test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: arduino/setup-protoc@v2
+        with:
+          version: "23.1"
+          repo-token: ${{ secrets.DEV_GITHUB_TOKEN }}
+      - name: Lint (Linux only)
+        if: runner.os == 'Linux'
+        run: |
+          cd rust
+          cargo clippy -r --verbose -- -D warnings # elevates warnings to errors
+      - name: Build
+        run: |
+          cd rust
+          cargo build --release --verbose --features "strict"
+      - name: Test
+        run: |
+          if [ $RUNNER_OS = 'macOS' ]; then
```

Comment:
I suggest either combining all 3 platforms into one Matrix or splitting all 3, but not mixing and matching 2+1. You even have weird things like "if windows" as a step prerequisite under the Windows only job.